test/functional/test_framework/test_framework.py

```diff
@@ -95,6 +95,7 @@ def __init__(self):
         self.nodes = []
         self.network_thread = None
         self.rpc_timeout = 60  # Wait for up to 60 seconds for the RPC server to respond
+        self.use_rpc_sync = False
```

Comment:
Curious, as the improvement is mostly gained by bypassing Poisson feature, why don't disable it via parameter e.,g,`-enablepoisson=0`, which would be allowed only in regtest? In this case, nodes behave closer to a real scenario, we have cleaner logs as txs are exchanged via P2P only (no race condition), would work even if `sync_mempool` hasn't been called and no need to maintain a custom tx exchange function.

Reply:
@kostyantyn Good point! Done in

* net: Send txs without delay on regtest #15881